mii/method_table.py

```diff
@@ -0,0 +1,158 @@
+from transformers import Conversation
+
+from mii.constants import Tasks
+from mii.grpc_related.proto import modelresponse_pb2
+from mii.utils import kwarg_dict_to_proto, unpack_proto_query_kwargs
+from mii.models.utils import ImageResponse
+
+
+def single_string_request_to_proto(request_dict, **query_kwargs):
+    return modelresponse_pb2.SingleStringRequest(
+        request=request_dict['query'],
+        query_kwargs=kwarg_dict_to_proto(query_kwargs))
+
+
+def single_string_response_to_proto(response, time_taken, model_time_taken):
+    return modelresponse_pb2.SingleStringReply(response=f"{response}",
+                                               time_taken=time_taken,
+                                               model_time_taken=model_time_taken)
+
+
+def multi_string_request_to_proto(request_dict, **query_kwargs):
+    return modelresponse_pb2.MultiStringRequest(
+        request=request_dict['query'] if isinstance(request_dict['query'],
+                                                    list) else [request_dict['query']],
+        query_kwargs=kwarg_dict_to_proto(query_kwargs))
+
+
+def proto_request_to_single_input(request):
+    args = (request.request, )
+    kwargs = unpack_proto_query_kwargs(request.query_kwargs)
+    return args, kwargs
+
+
+def proto_request_to_list(request):
+    args = ([r for r in request.request], )
+    kwargs = unpack_proto_query_kwargs(request.query_kwargs)
+    return args, kwargs
+
+
+def text_generation_pack_response_to_proto(response, time_taken, model_time_taken):
+    text_responses = []
+    for response in response:
+        text = response[0]['generated_text']
+        text_responses.append(text)
+
+    return modelresponse_pb2.MultiStringReply(response=text_responses,
+                                              time_taken=time_taken,
+                                              model_time_taken=model_time_taken)
+
+
+def question_answering_unpack_request_from_proto(request):
+    kwargs = unpack_proto_query_kwargs(request.query_kwargs)
+    kwargs["question"] = request.question
+    kwargs["context"] = request.context
+    return (), kwargs
```

Comment:
```suggestion
    args = ()
    return args, kwargs
```

I think this makes it more clear what we are returning from these functions. Type hints for the return types (or even just comments) may also be useful for future developers adding new tasks.

Reply:
I will change it as you suggested. For type hints, let me discuss in the response to other comments.